tests/unit/kernel/MaternKernel_unittest.cc

```diff
@@ -0,0 +1,190 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/kernel/MaternKernel.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/util/zip_iterator.h>
+
+using namespace shogun;
+
+TEST(MaternKernelTest_1_2, test_kernel_matrix)
+{
+	constexpr float64_t width = 2.0;
+	constexpr float64_t nu = 0.5;
+	constexpr int32_t cache_size = 10;
+
+	SGMatrix<float64_t> matrix{{0, 1, 2}, {3, 4, 5}};
+	SGMatrix<float64_t> expected_kernel_matrix{{1., 0.07441661},
+	                                           {0.07441661, 1.}};
+
+	auto features = std::make_shared<DenseFeatures<float64_t>>(matrix);
+
+	auto kernel = std::make_shared<MaternKernel>(
+	    features, features, cache_size, width, nu);
+	SGMatrix<float64_t> computed_kernel_matrix = kernel->get_kernel_matrix();
+
+	for (const auto & [ result, reference ] :
+	     zip_iterator(computed_kernel_matrix, expected_kernel_matrix))
```

Comment:
very nice this zip iterator!

Reply:
Indeed! One of these days I need to fix it to work with temporary values, so we can have
```cpp
for (const auto& [X, y]: DotIterator(feat), label) {
...
}
```

Reply:
yes that'd be ace!

Reply:
maybe make an issue for this? a student might pick it up?